src/data_structures/interval_tree.rs

```diff
@@ -272,13 +255,24 @@ fn swap_interval_data<N, D>(node_1: &mut Node<N, D>, node_2: &mut Node<N, D>) {
 }
 
 fn intersect<N: Ord>(range_1: &Range<N>, range_2: &Range<N>) -> bool {
-    range_1.end > range_2.start && range_1.start < range_2.end
+    range_1.start < range_1.end && range_2.start < range_2.end &&
```

Comment:
And about that newtype. Not a lot of experience here, but it seems that if we wrap it like this
```rust
pub struct Interval<N>(Range<N>);
```
We can no longer access the `Range` fields `start` and `end`. They are accessible using additional methods implemented on the `Interval` struct. But as far as I can see there are then 2 ways to create an `Interval`:
`Interval::new(10..100).unwrap()`
`Interval(10..100)`
The second method does not use our Interval::new and the range is not validated.